isites_migration/utils.py

```diff
@@ -0,0 +1,348 @@
+            try:
+                file_repository = FileRepository.objects.select_related('storage_node').only(
+                    'file_repository_id', 'storage_node'
+                ).get(file_repository_id=file_repository_id)
+                _export_file_repository(file_repository, keyword, topic_title)
+            except FileRepository.DoesNotExist:
+                logger.info("FileRepository does not exist for %s", file_repository_id)
+                continue
+
+            _export_topic_text(topic, keyword, topic_title)
+
+        zip_path_index = len(settings.EXPORT_DIR) + 1
+        keyword_export_path = os.path.join(settings.EXPORT_DIR, settings.CANVAS_IMPORT_FOLDER_PREFIX + keyword)
+        zip_filename = os.path.join(settings.EXPORT_DIR, "%s%s.zip" % (settings.CANVAS_IMPORT_FOLDER_PREFIX, keyword))
+        z_file = zipfile.ZipFile(zip_filename, 'w')
+        for root, dirs, files in os.walk(keyword_export_path):
+            for file in files:
+                file_path = os.path.join(root, file)
+                z_file.write(file_path, file_path[zip_path_index:])
+        z_file.close()
+        shutil.rmtree(keyword_export_path)
+
+        export_key = Key(s3_bucket)
+        export_key.key = "%s.zip" % keyword
+        export_key.set_metadata('Content-Type', 'application/zip')
+        export_key.set_contents_from_filename(zip_filename)
+        logger.info("Uploaded file export for keyword %s to S3 Key %s", keyword, export_key.key)
+
+        os.remove(zip_filename)
+
+        logger.info("Finished exporting files for keyword %s to S3 bucket %s", keyword, s3_bucket.name)
+    except Exception:
+        message = "Failed to complete file export for keyword %s", keyword
+        logger.exception(message)
+        raise RuntimeError(message)
+
+
+def import_files(keyword, canvas_course_id):
+    try:
+        progress_url = create_canvas_content_migration(keyword, canvas_course_id)
+
+        workflow_state = 'processing'
+        while workflow_state not in ['completed', 'failed']:
+            time.sleep(2)
+            progress = SDK_CONTEXT.session.request('GET', progress_url).json()
+            workflow_state = progress['workflow_state']
+            if workflow_state == 'completed':
+                lock_canvas_folder(canvas_course_id, settings.CANVAS_IMPORT_FOLDER_PREFIX + keyword)
+    except Exception:
+        message = "Failed to complete file import for keyword %s", keyword
+        logger.exception(message)
+        raise RuntimeError(message)
+
+    return workflow_state
+
+
+def create_canvas_content_migration(keyword, canvas_course_id):
+    try:
+        root_folder = _get_root_folder_for_canvas_course(canvas_course_id)
+        export_file_url = _get_export_s3_url(keyword)
+        logger.info("Importing iSites file export from %s to Canvas course %s", export_file_url, canvas_course_id)
+        response = content_migrations.create_content_migration_courses(
```

Comment:
Might there be a security/privacy issue with the fact that we're putting iSites content into a public space to upload it to Canvas?